pkg/local_workflows/whoami_workflow_test.go

```diff
@@ -0,0 +1,265 @@
+package localworkflows
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/snyk/go-application-framework/pkg/configuration"
+	"github.com/snyk/go-application-framework/pkg/mocks"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// roundTripFn
+type roundTripFn func(req *http.Request) *http.Response
+
+// RoundTrip
+func (f roundTripFn) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req), nil
+}
+
+// return *http.Client with Transport replaced to avoid making real calls
+func newTestClient(fn roundTripFn) *http.Client {
+	return &http.Client{
+		Transport: roundTripFn(fn),
+	}
+}
+
+func Test_WhoAmI_whoAmIWorkflowEntryPoint_requireExperimentalFlag(t *testing.T) {
+	// setup
+	logger := log.New(os.Stderr, "test", 0)
+	config := configuration.New()
+	config.Set("experimental", false)
+
+	// setup mocks
+	ctrl := gomock.NewController(t)
+	networkAccessMock := mocks.NewMockNetworkAccess(ctrl)
+	invocationContextMock := mocks.NewMockInvocationContext(ctrl)
+
+	// setup invocation context
+	invocationContextMock.EXPECT().GetConfiguration().Return(config)
+	invocationContextMock.EXPECT().GetLogger().Return(logger)
+	invocationContextMock.EXPECT().GetNetworkAccess().Return(networkAccessMock)
+	networkAccessMock.EXPECT().GetHttpClient().Return(http.DefaultClient).AnyTimes()
+
+	expectedError := errors.New("set `--experimental` flag to enable whoAmI command")
+
+	// run test
+	_, err := whoAmIWorkflowEntryPoint(invocationContextMock, nil)
+	assert.Equal(t, expectedError.Error(), err.Error())
+}
+
+func Test_WhoAmI_whoAmIWorkflowEntryPoint_happyPath(t *testing.T) {
+	// setup
+	logger := log.New(os.Stderr, "test", 0)
+	config := configuration.New()
+	config.Set("experimental", true)
+
+	payload := `{
```

Comment:
nitpick: from a maintainability aspect it is good not to share knowledge if not necessary, this means you could consider using the api contract data structure to create the json payload.